pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -1173,47 +1186,45 @@ func (r *ReconcileClusterDeployment) ensureManagedDNSZoneDeleted(cd *hivev1.Clus
 }
 
 func (r *ReconcileClusterDeployment) syncDeletedClusterDeployment(cd *hivev1.ClusterDeployment, cdLog log.FieldLogger) (reconcile.Result, error) {
+	switch _, relocateStatus, err := controllerutils.IsRelocating(cd); {
+	case err != nil:
+		cdLog.WithError(err).Error("could not determine relocate status")
+		return reconcile.Result{}, errors.Wrap(err, "could not determine relocate status")
+	case relocateStatus == hivev1.RelocateComplete:
+		cdLog.Info("clusterdeployment relocated, removing finalizer")
+		err := r.removeClusterDeploymentFinalizer(cd, cdLog)
+		if err != nil {
+			cdLog.WithError(err).Log(controllerutils.LogLevel(err), "error removing finalizer")
+		}
+		return reconcile.Result{}, err
+	}
```

Comment:
Can you clarify why we only case for relocation complete? It feels really dangerous to move past this block if we're in any other relocating state.

Reply:
I should have cases for the other relocating states, too. If it is outgoing or incoming, then the clusterdeployment controller should wait for the clusterrelocate controller to clear the state prior to deprovisioning. Good catch.